src/MagnumPlugins/AssimpImporter/Test/AssimpImporterTest.cpp

```diff
@@ -287,6 +349,690 @@ void AssimpImporterTest::openDataFailed() {
+        CORRADE_COMPARE(animation->trackResultType(1), AnimationTrackType::Vector3);
+        CORRADE_COMPARE(animation->trackTargetType(1), AnimationTrackTargetType::Translation3D);
+        CORRADE_COMPARE(animation->trackTarget(1), 1);
+        Animation::TrackView<const Float, const Vector3> translation = animation->track<Vector3>(1);
+        CORRADE_COMPARE(translation.interpolation(), Animation::Interpolation::Linear);
+        CORRADE_COMPARE(translation.before(), Animation::Extrapolation::Constant);
+        CORRADE_COMPARE(translation.after(), Animation::Extrapolation::Constant);
+        constexpr Vector3 translationData[]{
+            Vector3::yAxis(0.0f),
+            Vector3::yAxis(2.5f),
+            Vector3::yAxis(2.5f),
+            Vector3::yAxis(0.0f)
+        };
+        CORRADE_COMPARE_AS(translation.keys(), (Containers::StridedArrayView1D<const Float>{translationScalingKeys}), TestSuite::Compare::Container);
+        CORRADE_COMPARE_AS(translation.values(), (Containers::StridedArrayView1D<const Vector3>{translationData}), TestSuite::Compare::Container);
+        CORRADE_COMPARE(translation.at(1.5f), Vector3::yAxis(2.5f));
+
+        /* Scaling, linearly interpolated, sharing keys with translation */
+        CORRADE_COMPARE(animation->trackType(2), AnimationTrackType::Vector3);
+        CORRADE_COMPARE(animation->trackResultType(2), AnimationTrackType::Vector3);
+        CORRADE_COMPARE(animation->trackTargetType(2), AnimationTrackTargetType::Scaling3D);
+        CORRADE_COMPARE(animation->trackTarget(2), 2);
+        Animation::TrackView<const Float, const Vector3> scaling = animation->track<Vector3>(2);
+        CORRADE_COMPARE(scaling.interpolation(), Animation::Interpolation::Linear);
+        CORRADE_COMPARE(scaling.before(), Animation::Extrapolation::Constant);
+        CORRADE_COMPARE(scaling.after(), Animation::Extrapolation::Constant);
+        constexpr Vector3 scalingData[]{
+            Vector3{1.0f},
+            Vector3::zScale(5.0f),
+            Vector3::zScale(6.0f),
+            Vector3(1.0f),
+        };
+        CORRADE_COMPARE_AS(scaling.keys(), (Containers::StridedArrayView1D<const Float>{translationScalingKeys}), TestSuite::Compare::Container);
+        CORRADE_COMPARE_AS(scaling.values(), (Containers::StridedArrayView1D<const Vector3>{scalingData}), TestSuite::Compare::Container);
+        CORRADE_COMPARE(scaling.at(1.5f), Vector3::zScale(5.2f));
+    }
+}
+
+void AssimpImporterTest::animationGltfNoScene() {
+    /* Assimp refuses to import glTF animations if the file
+       has no scenes. This reuses the TinyGltfImporter test
+       files, not the corrected ones used by other tests. */
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("AssimpImporter");
+    CORRADE_VERIFY(importer->openFile(Utility::Directory::join(TINYGLTFIMPORTER_TEST_DIR,
+        "animation.gltf")));
+
+    CORRADE_COMPARE(importer->animationCount(), 0);
```

Comment:
I'd do this instead so it's clear that this is a known importer failure (and when Assimp eventually fixes itself, the test will say "unexpected pass" instead of "fail":

```cpp
    /* This reuses the TinyGltfImporter test files, not the corrected ones used by other tests. */
    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("AssimpImporter");
    CORRADE_VERIFY(importer->openFile(Utility::Directory::join(TINYGLTFIMPORTER_TEST_DIR,
        "animation.gltf")));

    CORRADE_EXPECT_FAIL("Assimp refuses to import glTF animations if the file has no scenes.");
    CORRADE_COMPARE(importer->animationCount(), 3);
```

Reply:
Done